tests/shared/src/test/scala/scala/meta/tests/tokenizers/BaseTokenizerCoverageSuite.scala

```diff
@@ -0,0 +1,331 @@
+package scala.meta.tests.tokenizers
+
+import scala.meta._
+import scala.meta.parsers.Parse
+
+import org.scalatest.FunSuite
+
+abstract class BaseTokenizerCoverageSuite extends FunSuite {
+  private val nl = "\n"
+  private val whiteOnBlack = fansi.Back.Black ++ fansi.Color.White
+  val dotty = dialects.Dotty
+
+  def checkNone[T <: Tree](source: String): Unit = {
+    val testName = fansi.Str(source).overlay(whiteOnBlack)
+    test(testName.toString) {
+      val tree = source.parse[Term].get.asInstanceOf[T]
+      val tokens = tree.children
+      assert(tokens.isEmpty)
+    }
+  }
+
+  /* It's not always possible to write the syntax of a tree node directly
+   * for example, Term.Repeated are wraped in Term.Apply: f(x: _*)
+   * We can write an arbitary function to extract a child node.
+   * ex: checkSome[Term.Param, Decl.Def](_.tokenEqual)("def f(x: A →=← 1): B")
+   *   R: Term.Param, T: Decl.Def, apply: extract first parameter
+   */
+  trait Projection[R, T] {
+    def apply(root: R): T
+  }
+
+  implicit val applyRepeated: Projection[Term.Apply, Term.Repeated] =
+    new Projection[Term.Apply, Term.Repeated] {
```

Comment:
Can we have a `projection[Term.Apply, Term.Repeated] { case Term.Apply(_, (r: Term.Repeated) :: Nil) => r }` helper to reduce `def apply(...)` boilerplate for these instances?